core_backend/app/routers/manage_language.py

```diff
@@ -0,0 +1,181 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..auth import get_current_fullaccess_user, get_current_readonly_user
+from ..db.db_models import (
+    LanguageDB,
+    delete_language_from_db,
+    get_default_language_from_db,
+    get_language_from_db,
+    get_list_of_languages_from_db,
+    is_language_name_unique,
+    save_language_to_db,
+    update_language_in_db,
+)
+from ..db.engine import get_async_session
+from ..schemas import AuthenticatedUser, LanguageBase, LanguageRetrieve
+from ..utils import setup_logger
+
+router = APIRouter(prefix="/language")
+logger = setup_logger()
+
+
+@router.post("/create", response_model=LanguageRetrieve)
+async def create_language(
+    language: LanguageBase,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve | None:
+    """
+    Create language endpoint. Upsert language to PG database
+    """
+    language.language_name = language.language_name.upper()
+    if not (await is_language_name_unique(language.language_name, asession)):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default is True:
+        await unset_default_language(asession)
+
+    language_db = await save_language_to_db(language, asession)
+
+    return _convert_language_record_to_schema(language_db)
+
+
+@router.put("/{language_id}/edit", response_model=LanguageRetrieve)
+async def edit_language(
+    language_id: int,
+    language: LanguageBase,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve | None:
+    """
+    Edit language endpoint.
+    """
+    old_language = await get_language_from_db(
+        language_id,
+        asession,
+    )
+
+    if not old_language:
+        raise HTTPException(
+            status_code=404, detail=f"Language id `{language_id}` not found"
+        )
+    language.language_name = language.language_name.upper()
+    if (old_language.language_name != language.language_name) and (
+        not (await is_language_name_unique(language.language_name, asession))
+    ):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default is True:
+        await unset_default_language(asession)
```

Comment:
Here also we can edit an existing default language to be not default, resulting in only non-default languages..

Reply:
Will fix that !

Reply:
Don't we need that information for the front-end? @Tanmay-97